sdk/common/src/main/java/io/opentelemetry/sdk/internal/GlobUtil.java

```diff
@@ -79,4 +65,42 @@ private static Pattern toRegexPattern(String globPattern) {
     }
     return Pattern.compile(patternBuilder.toString());
   }
+
+  private static class GlobPatternPredicate implements Predicate<String> {
+    private final String globPattern;
+    @Nullable private final Pattern pattern;
+
+    private GlobPatternPredicate(String globPattern) {
+      this.globPattern = globPattern;
+      // If globPattern contains '*' or '?', convert it to a regex and return corresponding
+      // predicate
+      Pattern pattern = null;
+      for (int i = 0; i < globPattern.length(); i++) {
+        char c = globPattern.charAt(i);
+        if (c == '*' || c == '?') {
+          pattern = toRegexPattern(globPattern);
```

Comment:
this method `toRegexPattern` only services this inner class now -- it should probably be moved into this static inner class to make it easier to reason about.

Reply:
With your other advice here, this utility class is now used by the outer class so keeping this as is